src/Polly.Core/Timeout/TimeoutValidator.cs

```diff
@@ -0,0 +1,15 @@
+namespace Polly.Timeout;
+
+internal static class TimeoutValidator
+{
+    internal static void ValidateTimeout(TimeSpan timeout)
+    {
+        if (timeout <= TimeSpan.Zero || timeout == System.Threading.Timeout.InfiniteTimeSpan)
```

Comment:
Should this be this?

```suggestion
        if (timeout <= TimeSpan.Zero && timeout != System.Threading.Timeout.InfiniteTimeSpan)
```

This seems to throw if you specify an infinite timeout but then the message says that that's a valid value.

Reply:
Yes indeed, fixed.